lib/util/Components.js

```diff
@@ -347,12 +347,13 @@ function componentRule(rule, context) {
         var isFunction = /Function/.test(node.type); // Functions
         var isMethod = node.parent && node.parent.type === 'MethodDefinition'; // Classes methods
         var isArgument = node.parent && node.parent.type === 'CallExpression'; // Arguments (callback, etc.)
+        var isJSX = node.parent && node.parent.type === 'JSXExpressionContainer';
```

Comment:
The problem is that functions inside a `JSXExpressionContainer` are bing mistaken for stateless components.

For example: `<button onClick={() => props.handleClick()}></button>`
where `{() => props.handleClick()}` is the`JSXExpressionContainer`